### ISSUE TITLE
Code base

### ISSUE THREAD

**Issue by MStumpp** [EvelynFan/FAU#1]
Any updates on when code will be made available?

**Comment by EvelynFan** [comment]
> Any updates on when code will be made available?

I am a little busy recently and still working on cleaning up the code. It shall be available in the end of April. Thanks for your attention.

**Comment by MStumpp** [comment]
Thanks for providing the code. Unfortunately, the link to the pretrained model seems broken.

**Comment by EvelynFan** [comment]
> Thanks for providing the code. Unfortunately, the link to the pretrained model seems broken.

Hi, recently I can't access the original server in our group due to the network issues. I have trained a new model and uploaded it to google drive. Please check it out :)

**Comment by MStumpp** [comment]
Excellent. Last question: What dataset has been used for training the model? BP4D and/or DISFA?

**Comment by EvelynFan** [comment]
> Excellent. Last question: What dataset has been used for training the model? BP4D and/or DISFA?

There are two models that have been trained using BP4D and DISFA, respectively. Since BP4D has 5 AU labels (AU6, 10, 12, 14, and 17), whereas DISFA has 12 AU labels, the output dimensions of the two models should be different.

**Comment by MStumpp** [comment]
> There are two models that have been trained using BP4D and DISFA, respectively. 

On Google drive I can see only one model. Shouldn't there be two separate models then?

> Since BP4D has 5 AU labels (AU6, 10, 12, 14, and 17), whereas DISFA has 12 AU labels, the output dimensions of the two models should be different.

Agree regarding DISFA, but the official BP4D dataset has approx. 30 AUs (AU01 - AU39) according to "BP4D-Spontaneous: a high-resolution spontaneous 3D dynamic facial expression database" paper.

**Comment by EvelynFan** [comment]
> > There are two models that have been trained using BP4D and DISFA, respectively.
> 
> On Google drive I can see only one model. Shouldn't there be two separate models then?
> 
> > Since BP4D has 5 AU labels (AU6, 10, 12, 14, and 17), whereas DISFA has 12 AU labels, the output dimensions of the two models should be different.
> 
> Agree regarding DISFA, but the official BP4D dataset has approx. 30 AUs (AU01 - AU39) according to "BP4D-Spontaneous: a high-resolution spontaneous 3D dynamic facial expression database" paper.

Yes. There should be two separate models and I'll upload the other one when I access the server in our group. Please try the model that has been uploaded first.

For the BP4D database, there are two types of AU annotation files: occurrence and intensity files. They only provide 5 AUs for intensity annotations. Please refer to FERA2015 challenge (https://ibug.doc.ic.ac.uk/resources/FERA15/) for details.

**Comment by EvelynFan** [comment]
> > > > There are two models that have been trained using BP4D and DISFA, respectively.
> > > 
> > > 
> > > On Google drive I can see only one model. Shouldn't there be two separate models then?
> > > > Since BP4D has 5 AU labels (AU6, 10, 12, 14, and 17), whereas DISFA has 12 AU labels, the output dimensions of the two models should be different.
> > > 
> > > 
> > > Agree regarding DISFA, but the official BP4D dataset has approx. 30 AUs (AU01 - AU39) according to "BP4D-Spontaneous: a high-resolution spontaneous 3D dynamic facial expression database" paper.
> > 
> > 
> > Yes. There should be two separate models and I'll upload the other one when I access the server in our group. Please try the model that has been uploaded first.
> > For the BP4D database, there are two types of AU annotation files: occurrence and intensity files. They only provide 5 AUs for intensity annotations. Please refer to FERA2015 challenge (https://ibug.doc.ic.ac.uk/resources/FERA15/) for details.
> 
> Hi, may we access to the pretrained model?

Hi, currently we have been implementing the Pytorch version. We will release the code and pre-trained model in the next few months.

**Comment by Mahdidrm** [comment]
Hello, could you please tell me if you have released the pre-trained model?

cordially
